### PR TITLE
Fix get_jobs method in Comparison Monitor

### PR DESCRIPTION
**Pull request by VMRuiz**
Fixes https://github.com/scrapinghub/spidermon/issues/442

### CODE REVIEW

**Comment by VMRuiz** [comment]
Force pushed branch again to remove incorrectly staged files

File: spidermon/contrib/scrapy/monitors/monitors.py [hunk 1]
```diff
@@ -557,26 +557,30 @@ def run(self, result):
...
-            _jobs = client.spider.jobs.list(
+        while True:
+            # API has a 1000 results limit
+            count = min(number_of_jobs - len(total_jobs), MAX_API_COUNT)
+            current_jobs = client.spider.jobs.list(
                 start=start,
                 state=states,
-                count=number_of_jobs,
+                count=count,
                 filters=dict(has_tag=tags) if tags else None,
```
Comment (curita) [inline spidermon/contrib/scrapy/monitors/monitors.py:572]: **issue (non-blocking):** This was already like this, so we might need to update it in another PR, but `filters=dict(has_tag=tags) if tags else None` doesn't do anything, at least in the version of `scrapinghub` I'm using (`2.4.0`). `has_tag=tags` does filter the jobs, though I'm not sure if that's backwards compatible.

**Review by curita — APPROVED** [review]
Changes look good to me!

Thanks for addressing these and the CI issue 🙇‍♀️

File: spidermon/contrib/scrapy/monitors/monitors.py (hunk 1 above)
Comment (VMRuiz) [reply spidermon/contrib/scrapy/monitors/monitors.py:572]: Is there any other issues about this?

Comment (VMRuiz) [reply spidermon/contrib/scrapy/monitors/monitors.py:572]: Is there any other issues about this?

Comment (curita) [reply spidermon/contrib/scrapy/monitors/monitors.py:572]: No that I know of, no! And that issue is not related to these changes. We should be good to merge 👍

**Pull request merged**